projects/graph/src/graph.py

```diff
@@ -130,6 +130,36 @@ def dfs(self, starting_vertex_id, target):
 
         print("Are you sure that's in there?")
 
+    def bfs(self, starting_vertex_id, target):
+        if target == starting_vertex_id:
+                print("but that's the same thing")
+        q = Queue()
+        q.enqueue(starting_vertex_id)
+
+        paths = {starting_vertex_id: [starting_vertex_id]}
+
+        while q.size() > 0:
+
+        # Add all edges of given vertex to queue
+            v = q.dequeue()
+            for neighbor in self.vertices[v]:
+                if neighbor not in paths:
+                # for i in self.vertices[queue[0]]:
+
+                # If vertex has not been visited, cache its path
+                # if i not in paths:
+                    paths[neighbor] = list(paths[v])
```

Comment:
By now we as a class had gone through the model solution, so when you get a chance, please take a moment to look at the difference between your solution and model solution.  While your solution may work, it is very memory intensive, since you have to maintain your path dictionary that creates a new list for every single neighbor of every single vertex that you visited, and the size of the dictionary can grow exponentially based on the average number of edges each vertex has in the graph.